Move most README content to index.rst

bhrutledge opened the pull request:
Note: I know this unexpected, and normally I would open an issue first, but the change was easy enough to just do, and I thought might facilitate discussion.

I thought of this while working on #644. By putting all of the source for the documentation in one location, it makes it easier to build and preview. More importantly (I think), it gives a canonical destination for users and contributors. I also like the simplicity of the README for rendering on GitHub and PyPI.

README preview: https://github.com/bhrutledge/twine/blob/move-readme/README.rst
Docs preview: https://twine-docs.bhrutledge.now.sh/

Prior art:

- https://github.com/pypa/pip/blob/master/README.rst
- https://github.com/pypa/packaging/blob/master/README.rst

Additional changes:

- Normalize to 4 spaces for indentation
- Fix some line-length issues
- Add README.rst to `doc8` linting
- Remove hack in `tox -e watch-docs`

bhrutledge commented on README.rst:
```diff
@@ -1,346 +1,29 @@
-.. rtd-inclusion-marker-do-not-remove
-
-Twine is `a utility`_ for `publishing`_ Python packages on `PyPI`_.
+Twine is a utility for `publishing`_ Python packages on `PyPI`_.
```
The `utility` link went to PyPI, which seemed redundant with the badge above.

bhrutledge commented on tox.ini:
```diff
@@ -35,9 +35,6 @@ deps =
 commands =
     sphinx-autobuild -W -b html -d {envtmpdir}/doctrees \
         {posargs:-H 127.0.0.1} \
-        # HACK: Detect changes in README.rst; results in superfluous builds
-        # Mitigated somewhat by ignoring dotfiles
-        -z . -i '.*' \
```
This was only necessary when `docs/index.rst` included `README.rst`.

sigmavirus24 commented on docs/index.rst:
```diff
@@ -1,25 +1,379 @@
+See `twine 338 <https://github.com/pypa/twine/issues/338>`_ for
+discussion and background.
+
+Options
```
Non-blocking nit: I just noticed that this seems this is either describing sub-command options or the subcommands or both so `Options` is a less than idea section title.

sigmavirus24 commented:
This looks good. Ideally the README is less overwhelming now and the docs serve their purpose better.

bhrutledge commented:
Thanks, @sigmavirus24. In https://github.com/pypa/twine/pull/656/commits/172a828091621555ce279bdfaf792770bb22a78a, I changed `Options` to `Commands`, and fixed broken list item indentation.

In https://github.com/pypa/twine/pull/656/commits/f4a4cd6b1aca1bb528fc84241c1469e17152fc11, I went a step further and shuffled some sections, prioritizing the command list above keyring, and pulling up "Configuration" in the outline.

https://twine-docs.bhrutledge.now.sh/

sigmavirus24 commented:
This all makes sense to me! Thanks again, @bhrutledge !
